File: legacy/code/Kt10/src/main/java/net/println/kt10/java/LazyNotThreadSafe.java
package net.println.kt10.java;

/**
 * Created by benny on 11/13/16.
 */
public class LazyNotThreadSafe {
    private static LazyNotThreadSafe INSTANCE;

    private LazyNotThreadSafe(){}

    public static LazyNotThreadSafe getInstance(){
        if(INSTANCE == null){
            INSTANCE = new LazyNotThreadSafe();
        }
        return INSTANCE;
    }
}

File: code/Kt10/src/main/kotlin/net/println/kt10/kotlin/LazyNotThreadSafe.kt
package net.println.kt10.kotlin

/**
 * Created by benny on 11/13/16.
 */
class LazyNotThreadSafe {

    companion object{
        val instance by lazy(LazyThreadSafetyMode.NONE) {
            LazyNotThreadSafe()
        }

        //下面是另一种等价的写法, 获取单例使用 get 方法
        private var instance2: LazyNotThreadSafe? = null

        fun get() : LazyNotThreadSafe {
            if(instance2 == null){
                instance2 = LazyNotThreadSafe()
            }
            return instance2!!
        }
    }
}